packages/react-percy-config/src/getWebpackSettings.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-console */
+
+import chalk from 'chalk';
+import path from 'path';
+
+class ConfigError extends Error {
+  constructor(message) {
+    super(message);
+
+    Error.stackTraceLimit = 0;
+    Error.captureStackTrace(this, ConfigError);
+
+    this.name = '';
+    this.message = chalk.bold.red(`Error: ${message}`);
+  }
+}
+
+function sanitizeCustomWebpackConfig(config = {}) {
+  if (typeof config !== 'object') {
+    throw new ConfigError(
+      `The \`webpack\` field in percy.config.js must be a plain object, but was "${typeof config}".`,
+    );
+  }
+
+  if (Array.isArray(config)) {
+    throw new ConfigError(
+      'The `webpack` field in percy.config.js must be a plain object, but was "array".',
+    );
+  }
+
+  // Do not allow users to customize the entry config
+  if (config.entry) {
+    console.log(chalk.bold.yellow('Warning: ignoring `webpack.entry` field in percy.config.js.'));
+    delete config.entry;
+  }
+
+  // Do not allow users to customize the output config
+  if (config.output) {
+    console.log(chalk.bold.yellow('Warning: ignoring `webpack.output` field in percy.config.js.'));
+    delete config.output;
+  }
+
+  // Move module.loaders to module.rules so we can properly merge with them
+  if (config.module) {
+    if (config.module.loaders) {
+      console.log(
+        chalk.bold.yellow(
+          'Warning: found `webpack.module.loaders` field in percy.config.js, but react-percy requires a webpack 3 compatible config.',
+        ),
+      );
+      console.log(
+        chalk.bold.yellow(
+          '`webpack.module.loaders` have been automatically moved to `webpack.module.rules`.',
+        ),
+      );
+      config.module.rules = config.module.loaders;
+      delete config.module.loaders;
+    }
+
+    if (config.module.preLoaders) {
+      console.log(
+        chalk.bold.yellow(
+          'Warning: found `webpack.module.preLoaders` field in percy.config.js, but react-percy requires a webpack 3 compatible config.',
```

Comment:
I think your messaging for the above rules where you indicate the unexpected field is being ignored is good.  This message should perhaps also inform the user `webpack.module.preLoaders` is being ignored?

Reply:
Updated:

![image](https://user-images.githubusercontent.com/733696/30515639-105e3178-9afa-11e7-9c1a-f56c1639300a.png)